kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -897,6 +900,39 @@ class ResidualBasedNewtonRaphsonStrategy
         KRATOS_CATCH("");
     }
 
+
+
+
+    void CalculateResidualNorm(
+        ModelPart& rModelPart,
+        TDataType& rResidualSolutionNorm,
+        DofsArrayType& rDofSet,
+        const TSystemVectorType& rb
+        )
+    {
+        // Initialize
+        TDataType residual_solution_norm = TDataType();
+        SizeType dof_num = 0;
+
+        // Auxiliar values
+        TDataType residual_dof_value = 0.0;
+        const auto it_dof_begin = rDofSet.begin();
+        const int number_of_dof = static_cast<int>(rDofSet.size());
+
+        #pragma omp parallel for firstprivate(residual_dof_value) reduction(+:residual_solution_norm, dof_num)
```

Comment:
You can use the parallel utils here

Reply:
Probably this should be computed in the ConveregenceCriteria instead of the NR

Reply:
yes yes all the implementation is provisional. I think that we should update the conv crit to be able to be used because I think not all of them have implemented the calculate residual in the same way

Reply:
Yes, I was thinking the same. To define this as an independent method, than later is called from PostCriteria, but it is public, so we can call it in places like this